rest-service/manager_rest/plugin_update/manager.py

```diff
@@ -0,0 +1,265 @@
+                                   ).items
+            if ex.status not in ExecutionState.END_STATES]
+
+        if real_active_updates:
+            raise ConflictError(
+                'There are plugins updates still active; the "force" flag '
+                'was used yet these updates have actual executions running '
+                'update IDs: {0}'.format(', '.join(
+                    [u.id for u in real_active_updates])))
+        else:
+            # the active updates aren't really active - either their
+            # executions were failed/cancelled, or the update failed at
+            # the finalizing stage.
+            # updating their states to failed and continuing.
+            for plugins_update in active_updates:
+                plugins_update.state = STATES.FAILED
+                self.sm.update(plugins_update)
+
+    @staticmethod
+    def _get_available_random_resource_id(prefix):
+        """Generates a random resource ID using the given prefix. It will
+        ensure that the ID doesn't pass 256 characters.
+        """
+        # Make sure not to pass 256 characters, but still use enough random
+        # letters/digits
+        suffix = str(uuid.uuid4()) + HIDDEN_BLUEPRINT_SUFFIX
+        cut_from_end = min(256 - len(suffix) - 1 - len(prefix), 0)
+        if cut_from_end == 0:
+            cut_from_end = len(prefix)
+        return '{0}-{1}'.format(prefix[:cut_from_end], suffix)
+
+    def _create_temp_blueprint_from(self, blueprint, temp_plan):
+        temp_blueprint_id = self._get_available_random_resource_id(
+            blueprint.id)
+        kwargs = {
+            'application_file_name': blueprint.main_file_name,
+            'blueprint_id': temp_blueprint_id,
+            'plan': temp_plan
+        }
+        # Make sure not to pass both private resource and visibility
+        visibility = blueprint.visibility
+        if visibility:
+            kwargs['visibility'] = visibility
+            kwargs['private_resource'] = None
+        else:
+            kwargs['visibility'] = None
+            kwargs['private_resource'] = blueprint.private_resource
+        return self.rm.publish_blueprint_from_plan(**kwargs)
+
+    def _stage_plugin_update(self, blueprint):
+        update_id = self._get_available_random_resource_id(blueprint.id)
+        plugins_update = models.PluginsUpdate(
+            id=update_id,
+            created_at=utils.get_formatted_timestamp())
+        plugins_update.set_blueprint(blueprint)
+        return self.sm.put(plugins_update)
+
+    def initiate_plugins_update(self, blueprint_id, force=False):
+        """Creates a temporary blueprint and executes the plugins update
+        workflow.
+        """
+
+        self.validate_no_active_updates_per_blueprint(blueprint_id, force)
+        blueprint = self.sm.get(models.Blueprint, blueprint_id)
+        temp_plan = self.get_reevaluated_plan(blueprint)
+
+        deployments_to_update = \
```

Comment:
use ( ) to avoid \ , don't do this always

Reply:
Not sure if what I did now is better. I can't use a generator, I need to save the list.

Reply:
what is the connection to '( )'?